virtual-hyperscript/index.js

```diff
@@ -40,16 +40,18 @@ function h(tagName, properties, children) {
         props.namespace = undefined;
     }
 
-    // fix cursor bug
+    // fix cursor bug #377
     if (tag === 'INPUT' &&
         !namespace &&
         props.hasOwnProperty('value') &&
         props.value !== undefined &&
         !isHook(props.value)
     ) {
+        if (typeof props.value === 'number' || props.value && props.value.valueOf !== Object.prototype.valueOf) props.value += ''
```

Comment:
Historically (https://github.com/Matt-Esch/virtual-dom/pull/154 - that much effort *just* for numbers) type coercion has been avoided.
My position is to continue strict type requirements.

Reply:
Ok, I just want to make sure there is understanding how "coercion is devil" works in this particular case. In #154 there is the opposite [opinion](https://github.com/Matt-Esch/virtual-dom/pull/154#issuecomment-124771390) as well.

 `h('input', {type: 'number', value: 123})` is natural expectation, and forcing coercion is putting dogma over common sense, also that contradicts to [W3C priority principle](https://www.w3.org/TR/html-design-principles/#priority-of-constituencies): _users over authors ... over theoretical purity_.

I observed many times how blind following dogmas brings any project to state of stagnation and death (look at [component org](https://github.com/component) - many contributors, no life), and I would not like that to happen to _virtual-dom_.

Reply:
see https://github.com/Matt-Esch/virtual-dom/pull/377#issuecomment-373896581